### PR TITLE
Added MAV_CMD_DO_SET_CAM_TRIGG_DIST to Mavlink commands

### PR DESCRIPTION
**Pull request by ptsneves**
@rmackay9 said in PR #6280:

> What I think might be better is to instead add support for receiving the MAV_CMD_DO_SET_CAM_TRIGG_DIST within a command-long (i.e. implement within the GCS_MAVLink file). Doing that would allow the ground station to start the distance based camera triggering in real-time not just from within a mission which is what we have now.

Based on his feedback I added this feature.

### CODE REVIEW

File: APMrover2/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -858,7 +858,11 @@ void GCS_MAVLINK_Rover::handleMessage(mavlink_message_t* msg)
             }
             result = MAV_RESULT_ACCEPTED;
             break;
-#endif  // CAMERA == ENABLED
+
+        case MAV_CMD_DO_SET_CAM_TRIGG_DIST:
+            copter.camera.set_trigger_distance(packet.param1);
```
Comment (WickedShell) [inline APMrover2/GCS_Mavlink.cpp]: There isn't a copter in rover. Same issue exists in plane. Please try compiling all the build targets if you change code in any of them.

**Comment by rmackay9** [comment]
Nice to see this change, though, thanks!
One other small thing, could you separate the changes into three commits?  we like to have commits separated by library or vehicle so that it's easier to cherry-pick them into branches.

**Comment by khancyr** [comment]
@ptsneves you are close to finish this PR ! 
You will need to squash you two commit together first: using git rebase - i master and -f option for example.
Then you need to split your commit into 3 commits : one for each vehicle type.
A git reset HEAD~ will rewing you git history from one commit wihout loose your change.
Then make a commit by vehicle. For the commit message, you need to follow ardupilot rules. The simplier is to look at previous commit message for changes on this file : https://github.com/ArduPilot/ardupilot/commits/master/ArduCopter/GCS_Mavlink.cpp . So an acceptable commit message will be Copter: Add MAV_CMD_DO_SET_CAM_TRIGG_DIST support !

**Comment by OXINARF** [comment]
@ptsneves Please re-read our contribution rules as your PRs aren't following them: http://ardupilot.org/dev/docs/submitting-patches-back-to-master.html We usually fix things for first-time contributors but we need contributors to learn the rules because unfortunately we don't have the man-power to correct every PR that comes in.

**Comment by ptsneves** [comment]
@OXINARF Sorry, i didn't notice there was an online page explaining the submission rules. I would risk saying nobody ever pointed me out that way, but I guess now I have no excuse ;). In my company we follow a similar workflow but I did not know if github handled force pushes well on pull requests. Now I know.
@khancyr I really think a lot of the GCS code needs to be refactored into the library as there seems to be quite a lot of common code. Did I miss something?
@rmackay9 Do you have TODO list?

**Comment by OXINARF** [comment]
@ptsneves That page is linked to when you open a PR :slightly_smiling_face: Forced pushes are fine. By the way, your commit messages are still wrong, it needs to start with library/vehicle/folder name - just check the Git history and you'll understand.

**Comment by rmackay9** [comment]
@ptsneves, re my to-do list, yes, I have one... or more like a few from different sources.

The top priority things are issues affecting AC3.5.  There's a list here on the issues list although there are a few things missing. https://github.com/ArduPilot/ardupilot/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22AC+3.5.0%22.

I have some other longer term projects that I'm doing as well like object avoidance and improving our MAVLink gimbal support.
I have been thinking about writing up a "getting involved" wiki page.  There are so many things that need doing (see the 800+ items on the issues list) but it can be a challenge for new people to know where to start because you want to pick something that is relatively small just to "get your feet wet".

**Comment by ptsneves** [comment]
@OXINARF :+1: Thanks for the hint.
@rmackay9 Some of the issues of the milestone look doable on my part. Expect new pull requests

**Comment by rmackay9** [comment]
Merged, thanks!

By the way, you might want to add this change for Sub as well...

**Pull request merged**